src/client.js

```diff
@@ -199,6 +199,34 @@ function getLoop54Client (endpoint, userId, apiKey, customHeaders) {
             return req;
         },
 
+        /**
+         * Used for performing getBasketRecommendations requests to the engine.
+         * @param {object[]} entities The set of entities in the basket to get recommendations for.
+         */
+        getBasketRecommendations: function (entities) {
+
+            var args = core.getOptionsAndCallback(arguments, 1, 4);
+            if (args.error) {
+                return core.returnError(args.error,args.callback);
+            }
+
+            var options = args.options ? args.options : {};
+            var callback = args.callback ? args.callback : null;
+
+            //validate entities
+            if (!Array.isArray(entities) || entities.filter(entity => entity.type && entity.id).length !== entities.length) {
+                return core.returnError("entities must be an array with atleast one object with properties \"type\" and \"id\".", callback);
```

Comment:
Somewhat misleading. Id say "entities must be a non-empty array where each item is an object with properties \"type\" and \"id\"."